org.eclipse.scanning.api/src/org/eclipse/scanning/api/malcolm/MalcolmTable.java

```diff
@@ -152,12 +152,18 @@ public boolean hasNext() {
 		
 	}
 
-	public Map<String, ArrayList<Object>> getTableData() {
+	public Map<String, List<Object>> getTableData() {
 		return tableData;
 	}
 
-	public void setTableData(Map<String, ArrayList<Object>> tableData) {
+	public void setTableData(Map<String, List<Object>> tableData) {
 		this.tableData = tableData;
+		// also sets the number of rows.
+		int numRows = tableData.values().iterator().next().size();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "numRows" which hides the field declared at line 30. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)